### ISSUE TITLE
Automated formatting tool for development

### ISSUE THREAD

**Issue by JBludau** [kokkos/pykokkos#103]
@NaderAlAwar and I already had a quick chat about this on slack. This issue would try to track the discussion on which automated formatting tool to use. I will go ahead an list the two we mentioned in the discussion.

- [pep8](https://peps.python.org/pep-0008/)
- [pfs/black](https://github.com/psf/black)

If anyone has any other suggestions or opinions/experiences with some of them please comment these below, we would like to incorporate these.

**Comment by tylerjereddy** [comment]
I would suggest **not** using a pre-commit hook--those are pretty annoying.

Many upstream projects have been hesitant to adopt `black` because of the way it formats mathematics and maybe arrays. There's a discussion about it in the scientific Python ecosystem discourse I think. See here, for example: https://discuss.scientific-python.org/t/how-to-format-mathematical-expressions/62

https://github.com/PyCQA/isort is popular for handling imports. Whatever you pick, the main priority should probably be just to remove all discussion about formatting within reason, and also it is important to make it trivial to replicate whatever the CI does for a formatting check directly on the command line. For that, you may to look at using rich-click CLI for offering a nice developer interface/help for common dev commands that are needed.

Seems low priority IMO though.

**Comment by JBludau** [comment]
yeah, but normally all the tools provide a facility to toggle them on and off within a file, I thought this would solve this problem

**Comment by JBludau** [comment]
> I would suggest not using a pre-commit hook--those are pretty annoying.

I actually have a format-on-close in my editor, turns out this works for most of my situations

**Comment by JBludau** [comment]
I will bring this up in the next meeting. Otherwise we can probably close this issue

**Comment by kennykos** [comment]
Closed by #315
